common/createdir.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2017 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "directorycleanuphandler.h"
+
+namespace softwarecontainer {
+
+class CreateDir {
+    LOG_DECLARE_CLASS_CONTEXT("CD", "Create Directory");
+public:
+    ~CreateDir();
+
+    /**
+     * @brief createDirectory creates a directory according to given path.
+     * The directory will be removed when the object is destroyed or rollBack method is called.
+     *
+     * @param path Path of directory to be created
+     * @return true on success, false on failure
+     */
+    bool createDirectory(const std::string path);
+
+    /**
+     * @brief rollBack removes all created directories.
+     *
+     * @return true on success, false on failure
+     */
+    bool rollBack();
+
+    /**
+     * @brief tempDir Creates a uniquely named directory according to templatePath.
+     *
+     * @warning The created directory will be destroyed when the object is destroyed or rollBack
+     * method is called.
+     *
+     * @param templ a template Path used to create the path of the temporary directory, including
+     *  XXXXXX which will be replaced with a unique ID for the temporary directory
+     *
+     * @throws SoftwareContainerError if template path is not appropriate.
+     * @return A string path pointing to the newly created temporary directory.
+     */
+    std::string tempDir(std::string templatePath);
```

Comment:
So "tempDir" should probably be named in a similar fashion as "createDirectory" is then? Otherwise the user will be forced to wonder why one says "create" and the other not, while both logically should create something.

And what do you mean with look into FileToolkitWithUndo? How would the user know that?